pyproject.toml

```diff
@@ -1,3 +1,77 @@
+[build-system]
+requires = ["setuptools", "setuptools-scm"]
+build-backend = "setuptools.build_meta"
+
+[project]
+authors = [
+    {name = "Darcy Mason and contributors", email = "darcymason@gmail.com"}
+]
+classifiers=[
+    "License :: OSI Approved :: MIT License",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Healthcare Industry",
+    "Intended Audience :: Science/Research",
+    "Development Status :: 5 - Production/Stable",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Operating System :: OS Independent",
+    "Topic :: Scientific/Engineering :: Medical Science Apps.",
+    "Topic :: Scientific/Engineering :: Physics",
+    "Topic :: Software Development :: Libraries"
+]
+dependencies = []
+description = "A pure Python package for reading and writing DICOM data"
+#  dynamic = ["version"]  uncomment for pydicom v3.0
+keywords = ["dicom, python, medical, imaging"]
+license = {text = "MIT"}
+name = "pydicom"
+readme = "README.md"
+requires-python = ">=3.7"
+version = "2.4.0dev0" # remove for v3.0
+
+
+[project.optional-dependencies]
+docs = [
+        "numpy",
+        "numpydoc",
+        "matplotlib",
+        "pillow",
```

Comment:
If we have pillow here, why not gdcm and libjpeg?

Reply:
Not sure, I just copied these over from setup.py.  I vaguely recall at one point when I tried to build docs locally, I found extra libraries that were needed.  Another thing to take a good look at for v3.0, I think.

Reply:
Ok, maybe it was needed for some of the examples...
I see that you have brought back `setup.py` - is that because of the breaking change?